From: "@KIC-8462852" <>
Date: Tue, 18 Jan 2022 11:57:01 -0500
Subject: Add ARC override for policy "quarantine"

Origin: other, https://github.com/trusteddomainproject/OpenDMARC/files/6697440/opendmarc-arc-overwrite-for-quarantines-patch.txt
Bug: https://github.com/trusteddomainproject/OpenDMARC/issues/24
---
 opendmarc/opendmarc.c | 29 +++++++++++++++++------------
 1 file changed, 17 insertions(+), 12 deletions(-)

diff --git a/opendmarc/opendmarc.c b/opendmarc/opendmarc.c
index be3d496..9317817 100644
--- a/opendmarc/opendmarc.c
+++ b/opendmarc/opendmarc.c
@@ -3637,16 +3637,7 @@ mlfi_eom(SMFICTX *ctx)
 		    conf->conf_holdquarantinedmessages &&
 		    random() % 100 < pct)
 		{
-			snprintf(replybuf, sizeof replybuf,
-			         "quarantined by DMARC policy for %s",
-			         pdomain);
-
-			status = smfi_quarantine(ctx, replybuf);
-			if (status != MI_SUCCESS && conf->conf_dolog)
-			{
-				syslog(LOG_ERR, "%s: smfi_quarantine() failed",
-				       dfc->mctx_jobid);
-			}
+			/* quarantine will be deferred to after the ARC policy eval */
 
 			ret = SMFIS_ACCEPT;
 			result = DMARC_RESULT_QUARANTINE;
@@ -3683,7 +3674,7 @@ mlfi_eom(SMFICTX *ctx)
 	**  arc.chain to assist with administrative debugging.
 	*/
 
-	if (result == DMARC_RESULT_REJECT &&
+	if ((result == DMARC_RESULT_REJECT || result == DMARC_RESULT_QUARANTINE) &&
 	    dfc->mctx_arcpass == ARES_RESULT_PASS &&
 	    dfc->mctx_arcpolicypass != DMARC_ARC_POLICY_RESULT_PASS &&
 	    conf->conf_dolog)
@@ -3693,7 +3684,7 @@ mlfi_eom(SMFICTX *ctx)
 		       dfc->mctx_jobid);
 	}
 
-	if (result == DMARC_RESULT_REJECT &&
+	if ((result == DMARC_RESULT_REJECT || result == DMARC_RESULT_QUARANTINE) &&
 	    dfc->mctx_arcpolicypass == DMARC_ARC_POLICY_RESULT_PASS)
 	{
 		ret = SMFIS_ACCEPT;
@@ -3706,6 +3697,20 @@ mlfi_eom(SMFICTX *ctx)
 		}
 	}
 
+	if (result == DMARC_RESULT_QUARANTINE)
+	{
+		snprintf(replybuf, sizeof replybuf,
+		         "quarantined by DMARC policy for %s",
+		         pdomain);
+
+		status = smfi_quarantine(ctx, replybuf);
+		if (status != MI_SUCCESS && conf->conf_dolog)
+		{
+			syslog(LOG_ERR, "%s: smfi_quarantine() failed",
+			       dfc->mctx_jobid);
+		}
+	}
+
 	/*
  	**  Append arc override to historyfile.  The format 
 	**
