From: Maximilian Eschenbacher <maximilian@eschenbacher.email>
Date: Thu, 16 Mar 2023 11:17:39 +0100
Subject: opendmarc-check: print correct domain in loop

Bug: https://github.com/trusteddomainproject/OpenDMARC/pull/209
---
 opendmarc/opendmarc-check.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/opendmarc/opendmarc-check.c b/opendmarc/opendmarc-check.c
index c569cd4..48f0fc3 100644
--- a/opendmarc/opendmarc-check.c
+++ b/opendmarc/opendmarc-check.c
@@ -223,7 +223,7 @@ main(int argc, char **argv)
 		rua = opendmarc_policy_fetch_rua(dmarc, NULL, 0, 1);
 		ruf = opendmarc_policy_fetch_ruf(dmarc, NULL, 0, 1);
 
-		fprintf(stdout, "DMARC record for %s:\n", argv[1]);
+		fprintf(stdout, "DMARC record for %s:\n", argv[c]);
 		fprintf(stdout, "\tSample percentage: %d\n", pct);
 		fprintf(stdout, "\tDKIM alignment: %s\n", adkim);
 		fprintf(stdout, "\tSPF alignment: %s\n", aspf);
