From: Scott Kitterman <scott@kitterman.com>
Date: Mon, 23 Dec 2019 11:12:36 -0500
Subject: ticket204

Bug: https://sourceforge.net/p/opendmarc/tickets/204/
---
 reports/opendmarc-import.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/reports/opendmarc-import.in b/reports/opendmarc-import.in
index 3a2f404..0169c9e 100755
--- a/reports/opendmarc-import.in
+++ b/reports/opendmarc-import.in
@@ -649,7 +649,7 @@ while (<$inputfh>)
 				}
 
 	  case "from"		{
-					$fdomain = $value;
+					$fdomain = lc($value);
 				}
 
 	  case "job"		{
@@ -691,7 +691,7 @@ while (<$inputfh>)
 				}
 
 	  case "mfrom"		{
-					$envdomain = $value;
+					$envdomain = lc($value);
 				}
 
 	  case "p"		{
@@ -703,7 +703,7 @@ while (<$inputfh>)
 				}
 
 	  case "pdomain"	{
-					$pdomain = $value;
+					$pdomain = lc($value);
 				}
 
 	  case "policy"		{
